ios/gamma/split-view/RNSSplitViewScreenController.swift

```diff
@@ -69,16 +77,64 @@ public class RNSSplitViewScreenController: UIViewController {
 
   // MARK: Layout
 
+  ///
+  /// @brief This method is overridden to extract the value to which we're transitioning
+  /// and attach the DisplayLink to track frame updates on the presentation layer.
+  ///
+  public override func viewWillTransition(
+    to size: CGSize,
+    with coordinator: any UIViewControllerTransitionCoordinator
+  ) {
+    super.viewWillTransition(to: size, with: coordinator)
+
+    transitioningToSize = size
+    isTransitioning = true
+
+    if displayLink == nil {
+      displayLink = CADisplayLink(target: self, selector: #selector(checkTransitionProgress))
+      displayLink?.add(to: .main, forMode: .common)
+    }
+  }
+
+  ///
+  /// @brief This method is responsible for tracking animation frames and requests layout
+  /// which will synchronize ShadowNode size with the animation frame size.
+  ///
+  @objc
+  private func checkTransitionProgress() {
+    guard let targetSize = transitioningToSize else { return }
+
+    if let currentFrame = view.layer.presentation()?.frame {
+      let currentSize = currentFrame.size
+
+      if abs(currentSize.width - targetSize.width) < viewWidthEpsilon
+        && abs(currentSize.height - targetSize.height) < viewWidthEpsilon
+      {
+        displayLink?.invalidate()
+        displayLink = nil
+        transitioningToSize = nil
+      }
+      animationFrame = currentFrame
+      view.setNeedsLayout()
```

Comment:
Do we really need to go through such pipeline? Maybe we just should trigger the shadow state update directly here? Not sure if messing with native layout system during native transition is a good idea but I may be missing something here.